### ISSUE TITLE
example

### ISSUE THREAD

**Issue by juliangruber** [karissa/hyperdrive-archive-swarm#10]
The currently released version is missing two dependencies (`inherits` and `pump`). I added them and also added an `example.js` which borrows code from the README so a developer can quickly run the important parts of the code.

With a fresh set of dependencies, I'm now getting this error:

```
/Users/julian/dev/karissa/hyperdrive-archive-swarm/node_modules/discovery-swarm/node_modules/to-buffer/index.js:6
  throw new Error('Input should be a buffer or a string')
  ^

Error: Input should be a buffer or a string
    at toBuffer (/Users/julian/dev/karissa/hyperdrive-archive-swarm/node_modules/discovery-swarm/node_modules/to-buffer/index.js:6:9)
    at Swarm.join (/Users/julian/dev/karissa/hyperdrive-archive-swarm/node_modules/discovery-swarm/index.js:113:10)
    at Swarm.<anonymous> (/Users/julian/dev/karissa/hyperdrive-archive-swarm/index.js:65:11)
    at emitNone (events.js:72:20)
    at Swarm.emit (events.js:166:7)
    at UTP.onutplisten (/Users/julian/dev/karissa/hyperdrive-archive-swarm/node_modules/discovery-swarm/index.js:423:10)
    at emitNone (events.js:67:13)
    at UTP.emit (events.js:166:7)
    at /Users/julian/dev/karissa/hyperdrive-archive-swarm/node_modules/discovery-swarm/node_modules/utp-native/index.js:415:15
    at nextTickCallbackWith0Args (node.js:420:9)
```

Which suggests that `archive.discoveryKey` is null. Do you have an idea what's going on here?

**Comment by juliangruber** [comment]
Also, I wanted to do a quick npm release with the missing deps but wasn't sure what semver bump would be most appropriate.

**Comment by mafintosh** [comment]
@juliangruber you are getting that error because the archive isn't finalised and isn't live. then it wont have a key to share yet.

**Comment by mafintosh** [comment]
@juliangruber i think the 3.0.0 on npm is actually master of this repo although it was tagged a bit before. So a patch bump would then be ok.

**Comment by juliangruber** [comment]
thanks @mafintosh! published as 3.0.1 and created a pull request to fix the example
